### PR TITLE
feat: add pre commit config

### PR DESCRIPTION
**Pull request by jandom**
Description: 

What problem does this PR solve? It introduces minimal level of automated code formatting. Code formatting is good because it makes all the code look and feel the same. Differences of course remain but the code is "tidier". 

Changes made in this Pull Request:
 - create a pre-commit config file to automatically fix simple file formatting problems
 - updated a Github Actions workflow to run the linter check, disabled for now
 - PR with automated code changes

PR Checklist
------------
 - [x] Tests?
 - [ ] ~~Docs?~~ There were README-like docs in the repo but I haven't found anything where this would belong, happy to change if pointed in a direction
 - [x] CHANGELOG updated?
 - [ ] ~~Issue raised/referenced?~~ Not applicable, there was no issue raised 

## Developers certificate of origin
- [x] I certify that this contribution is covered by the LGPLv2.1+ license as defined in our [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

----
:books: Documentation preview :books:: https://mdanalysis--4321.org.readthedocs.build/en/4321/

### CODE REVIEW

File: .github/workflows/gh-ci.yaml [hunk 1]
```diff
@@ -124,6 +124,18 @@ jobs:
         fail_ci_if_error: True
         verbose: True
 
+  lint:
```
Comment (IAlibay) [inline .github/workflows/gh-ci.yaml:127]: There's a separate yaml file for linter actions called linters.yaml.

**Review by IAlibay — COMMENTED** [review]
May I suggest you open an issue which pings the coredevs to discuss your plans here? It might be a tad bit contentious.

**Comment by orbeckst** [comment]
I am echoing @IAlibay here: given the level of discussion about black/blacken/etc you should probably start with an issue and then reference the issue here.

_Maybe_ there's a growing critical mass in favor of auto-formatting (look for issues related to black etc) but the way to create disruptive change in a project like this is to start with a discussion and convincing enough people that it's a good idea.

I think at a minimum you'd have to explain what the implications of the proposed change are — not everyone is using the same tools and ideas as you do and what maybe self-evident to you is not for others.

**Comment by RMeli** [comment]
There is an old issue here, where the discussion can be continued/restarted: https://github.com/MDAnalysis/mdanalysis/issues/2450

**Comment by jandom** [comment]
don't think i'll be able to complete this in the coming month or so, so i'm closing for better visibility. will hopefully re-open in the future.

**Pull request closed without merging**